hera_pspec/pspecdata.py

```diff
@@ -1721,6 +1906,30 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None, input_data_weight='identit
                         if verbose: print("  Computing and multiplying scalar...")
                         pv *= scalar * self.scalar_delay_adjustment(key1, key2, sampling=sampling)
 
+                    #Generate the covariance matrix if error bars provided
+                    if covariance:
+                        if verbose: print(" Building q_hat covariance...")
+                        cov_qv=np.zeros((self.Ntimes,
+                                         self.spw_Ndlys,
+                                         self.spw_Ndlys),dtype=complex)
+                        cov_pv=np.zeros_like(cov_qv)
+                        amat,bmat=np.meshgrid(range(self.spw_Ndlys),
+                                              range(self.spw_Ndlys))
+                        amat=amat.astype(int)
+                        bmat=bmat.astype(int)
```

Comment:
Is there a use case for not wanting the entire covariance matrix? In other words, will we ever just want a single element (i.e., the covariance between two specific bandpowers)? If not, then I would suggest that for code uniformity, we move some of this stuff with `amat` and `bmat` to the `cov_qv` function itself. This way, the function acts more like the functions that return `M` and `W`, handling the loops over `self.spw_Ndlys` internally and just returning the entire `self.spw_Ndlys` by `self.spw_Ndlys` matrix.

The time dimension could also be folded into the `cov_qv` function. That would make it more similar to the function that generates `qv` itself, which handles all the times at once.

Reply:
My primary reason for allowing individual elements was time since the einsum implementation runs pretty slowly and I imagined there might be situations where someone would only want diagonal elements. I will change this after replacing einsum with something faster.

Reply:
cov_q_hat now returns a Ntimes x Ndlys x Ndlys matrix, similar to q_hat.